sitestats/sitestats-tool/src/java/org/sakaiproject/sitestats/tool/entityproviders/SiteStatsEntityProvider.java

```diff
@@ -149,7 +149,11 @@ public Object getSampleEntity() {
         if (reportDef == null) {
             throw new EntityException("Report with id '" + reportId + "' doesn't exist.", "", HttpServletResponse.SC_BAD_REQUEST);
         }
-
+        
+        if (reportDef.getReportParams().getSiteId() == null || reportDef.getReportParams().getSiteId().length() == 0) {
```

Comment:
`StringUtils.isNotBlank(reportDef.getReportParams().getSiteId())` would be preferred here.